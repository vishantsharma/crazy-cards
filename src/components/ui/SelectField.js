import React from "react";

const SelectField = ({
  name,
  label,
  placeholder,
  value,
  options = [],
  optionKey = "id",
  optionText = "name",
  onChange,
  error,
  className
}) => {
  return (
    <>
      <label htmlFor={name}>{label}</label>
      {error ? (
      <div role="alert" className="caution">
        <span>{error}</span>
      </div>
    ) : null}
      <select
        name={name}
        id={name}
        value={value}
        onChange={onChange}
        className={className}
      >
        <option value="" disabled>
          {placeholder}
        </option>
        {options.map((option) => (
          <option key={option[optionKey]} value={option[optionKey]}>
            {option[optionText]}
          </option>
        ))}
      </select>
      </>
  );
};

export default SelectField;
