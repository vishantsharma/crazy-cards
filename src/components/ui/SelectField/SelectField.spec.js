import React from 'react';
import TestRenderer from 'react-test-renderer';
import SelectField from './SelectField';

describe('SelectField', () => {
    let props = {
        name: '',
        label: '',
        placeholder: '',
        value: '',
        options: [],
        optionKey: "id",
        optionText: "name",
        onChange: jest.fn().mockReturnValue(),
        error: '',
        className: ''
    }
    it('should render', () => {
        const component = TestRenderer.create(<SelectField {...props} />)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})