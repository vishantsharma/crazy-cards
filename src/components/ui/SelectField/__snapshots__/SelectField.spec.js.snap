// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField should render 1`] = `
Array [
  <label
    htmlFor=""
  >
    
  </label>,
  <select
    className=""
    id=""
    name=""
    onChange={[MockFunction]}
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      
    </option>
  </select>,
]
`;
